src/combiners/combine-latest.ts

```diff
@@ -13,10 +13,20 @@
 
 import { Observable } from "../types.js";
 
-export function combineLatest<T>(...os: Array<Observable<T>>): Observable<T[]> {
-  const hasValue = new Set();
-  const latestValue = (os.map(() => null) as any) as T[];
-  return new ReadableStream<T[]>({
+export function combineLatest<T1, T2>(
+  o1: Observable<T1>,
+  o2: Observable<T2>
+): Observable<[T1 | null, T2 | null]>;
```

Comment:
It doesn’t ever emit `null`. If one of the observables hasn't emitted a value yet, no value gets emitted. See tests :)

(Is this conflicting with RxJS?)

Reply:
I don't know if it conflicts. I'm mostly writing these docs by looking at the implementation so I just go it wrong 😄

Reply:
I think I wrote all these with the [RxJS documentation](https://rxjs-dev.firebaseapp.com/api/index/function/combineLatest) on-screen. I double-checked and this seems to align:

> To ensure output array has always the same length, combineLatest will actually wait for all input Observables to emit at least once

👏